python/unit-tests/test_unsigned_multiplier.py

```diff
@@ -0,0 +1,130 @@
+from assassyn.frontend import *
+from assassyn.backend import elaborate
+from assassyn import utils
+import assassyn
+import pytest
+import random
+
+#AIM: unsigned 32 bit multiplier: 32b*32b=64b
+#DATE: 2025/4/16
+
+# MulStage 1: multiply each bit of b
+class MulStage1(Module):
+    def __init__(self):
+        super().__init__(
+            ports={
+                'a': Port(Int(32)),
+                'b': Port(Int(32)),
+                'cnt': Port(Int(32)),
+            }
+        )
+
+    @module.combinational
+    def build(self, stage1_reg: Array):
+        a, b, cnt = self.pop_all_ports(True)
+        bit_num0 = cnt % Int(32)(32) #to avoid overflow
+        b_bit = (b>>bit_num0)&Int(32)(1) #to get the cnt-th bit from the right
+        b_bit = b_bit.bitcast(Int(1))
+        stage1_reg[0] = ((a * b_bit).bitcast(Int(32))).bitcast(Int(64)) #'a' multiply b[cnt-1]

```

Comment:
It is my bad. DO NOT use `a.bitcast(Int(1))`. USE `a[0:0]` slicing.

I guess your problem is because mismatched data type.